packages/file-storage/src/file-storage-s3.class.ts

```diff
@@ -20,12 +21,13 @@ import {
 } from './file-storage.class';
 
 export type FileStorageS3Setup = {
-  accessKeyId: string;
   bucket: string;
-  endpoint: string;
-  secretAccessKey?: string;
-  s3BucketEndpoint?: boolean;
   maxPayloadSize: number;
+  region: string;
+  credentials: {
+    accessKeyId: string;
+    secretAccessKey: string;
```

Comment:
removing endpoint is a change which is quite big as it involves updating environment variables for all apps that uses it. So instead let's allow a softer transition by allowing both (oneOf) endpoint and region parameter.
If endpoint is provided you extract the region from it.